pennylane/transforms/core/transform.py

```diff
@@ -26,40 +27,77 @@ def transform(
     is_informative=None,
     final_transform=False,
 ):
-    """The transform function is to be used to validate and dispatch a quantum transform on PennyLane objects (tape,
-    qfunc and Qnode). It can be used directly as a decorator on qfunc and qnodes.
+    """The transform function is to be used to validate and dispatch a quantum transform on PennyLane objects
+    (:class:`pennylane.tape.QuantumTape`, quantum function and :class:`pennylane.QNode`). After applying the function on
+    a valid quantum transform, the resulted function can be used on all the mentioned objects. The function can be used
+    directly as a decorator on quantum function and :class:`pennylane.QNode`.
 
     Args:
         quantum_transform (callable): A quantum transform is defined as a function that has the following requirements:
 
-            * A quantum transform is a function that takes a quantum tape as first input and returns a sequence of tapes
-              and a processing function.
+            * A quantum transform is a function that takes a :class:`pennylane.tape.QuantumTape` as first input and
+              returns a sequence of :class:`pennylane.tape.QuantumTape` and a processing function.
 
-            * The transform must have type hinting of the following form: my_quantum_transform(tape:
-              qml.tape.QuantumTape, ...) -> ( Sequence[qml.tape.QuantumTape], callable)
+            * The transform must have type hinting of the following form: ``my_quantum_transform(tape:
+              qml.tape.QuantumTape, ...) -> ( Sequence[qml.tape.QuantumTape], callable)``
```

Comment:
I usually don't mind but `callable` is a function and I don't want users copying without realizing.
```suggestion
              qml.tape.QuantumTape, ...) -> ( Sequence[qml.tape.QuantumTape], Callable)``
```